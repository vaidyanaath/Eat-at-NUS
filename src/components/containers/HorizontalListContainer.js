import React from 'react';
import {
    TouchableOpacity,
    View,
    Image,
    Text,
    StyleSheet,
} from 'react-native';

// import texts
import { RegularText } from '../../components/texts/RegularText';
import { SmallText } from '../../components/texts/SmallText';

// import colors
import { colors } from '../../assets/colors';

export const HorizontalListContainer = ({ item }) => (
    <TouchableOpacity style={cardStyles.button}>
        <View style={cardStyles.container}>
            <View style={cardStyles.topContainer}>
                <Image 
                    style={cardStyles.image} 
                    resizeMode='cover'
                    source={require('../../assets/images/food2.jpg')} 
                />
                
            </View>
            <View style={cardStyles.bottomContainer}>
                <View style={cardStyles.textContainer}>
                    <RegularText style={cardStyles.dishName}>
                    {item.name}
                    </RegularText>
                    <SmallText style={cardStyles.dishInfo}>
                    {item.id} Km away
                    </SmallText>

                </View>
                <View style={cardStyles.ratingBG}>
                    <Text style={cardStyles.stallRating}>
                        {item.rating}
                    </Text>
                </View>
            </View>
        </View>
    </TouchableOpacity>
);

  
const cardStyles = StyleSheet.create({
button: {
    flex: 1,
    marginVertical: 5,
    marginHorizontal: 10,
    activeOpacity: 0.8,
    // borderTopLeftRadius: 20,
    // borderTopRightRadius: 20,
    // borderRadius: 20,

    borderColor: colors.gray,
    borderWidth: 0,
    backgroundColor: colors.white,
    shadowColor: '#000',
    shadowOffset: { width: 5, height: 5 },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,
    elevation: 3,
    minHeight: 155,
    minWidth: 150,
},
container: {
    flex: 1,
    flexDirection: 'column',
    minWidth: 130,
},
topContainer: {
    flex: 1,
    // borderTopLeftRadius: 20,
    // borderTopRightRadius: 20,
    alignItems: 'center',
    justifyContent: 'flex-start',
    minHeight: 50,
    
},
image: {

    height: '95%',
    width: '100%',
    // borderRadius: 20,
    
},
bottomContainer: {
    flex: 1,
    // borderBottomRightRadius: 20,
    // borderBottomLeftRadius: 20,
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    flexDirection: 'row',
    paddingBottom: 5,
    paddingHorizontal: 10,
    // backgroundColor: "#2ba",
},
textContainer: {
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    flexDirection: 'column',
    // backgroundColor: "#abcdef",
},
dishName: {
    fontSize: 20,
},
dishInfo: {
    flex: 1,
    fontSize: 15,
},
stallRating: {
    fontSize: 13,
},
ratingBG: {
    alignSelf: 'flex-end',
    marginBottom: 5,
    paddingHorizontal: 4,
    borderRadius: 3,
    backgroundColor: '#FFB81C',
    
}
});