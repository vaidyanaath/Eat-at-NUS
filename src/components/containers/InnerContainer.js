import styled from 'styled-components/native';

export const InnerContainer = styled.View`
  flex: 1;
  width: 100%;
  align-items: center;
  justify-content: center;
`;
