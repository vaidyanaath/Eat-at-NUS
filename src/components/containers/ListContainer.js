import React from 'react';
import {
    TouchableOpacity,
    View,
    Image,
    Text,
    StyleSheet,
} from 'react-native';

// import texts
import { RegularText } from '../../components/texts/RegularText';

// import colors
import { colors } from '../../assets/colors';

export const ListContainer = ({ item, onPress }) => {
    return (
        <TouchableOpacity style={cardStyles.button} onPress={onPress}>
        <View style={cardStyles.container}>
            <View style={cardStyles.imageContainer}>
              <Image 
                  style={cardStyles.image} 
                  source={require('../../assets/images/food3.jpg')} 
              />
            </View>
            <View style={cardStyles.textContainer}>
                <RegularText style={cardStyles.stallName}>
                  {item.name}
                </RegularText>
                <RegularText style={cardStyles.stallDistance}>
                  {item.id} Km away
                </RegularText>
                
                <View style={cardStyles.ratingBG}>
                  <Text style={cardStyles.stallRating}>
                      {item.rating}
                  </Text>
                </View>
                
            </View>
        </View>
        </TouchableOpacity>
    );
}


  
const cardStyles = StyleSheet.create({
button: {
    flex: 1,
    height: 100,
    width: '98%',
    marginVertical: 5,
    activeOpacity: 0.8,
    borderRadius: 20,
    borderColor: colors.gray,
    borderWidth: 0,
    backgroundColor: colors.white,
    shadowColor: '#000',
    shadowOffset: { width: 5, height: 5 },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,
    elevation: 3,
},
container: {
    flex: 1,
    flexDirection: 'row',
},
imageContainer: {
    flex: 1,
    borderTopLeftRadius: 20,
    borderBottomLeftRadius: 20,
    // backgroundColor: "#fef342",
    maxWidth: 140,
    // width: 150,
},
image: {
    flex: 1,
    borderRadius: 20,
    resizeMode: 'cover',
    width: 125,
},
textContainer: {
    flex: 1,
    borderBottomRightRadius: 20,
    borderTopRightRadius: 20,
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    flexDirection: 'column',
    paddingVertical: 10,
    //backgroundColor: "#abcdef",
},
stallName: {
    fontSize: 20,
},
stallDistance: {
    flex: 2,
    fontSize: 15,
},
stallRating: {
    fontSize: 15,
},
ratingBG: {
    paddingHorizontal: 4,
    borderRadius: 3,
    backgroundColor: '#FFB81C',
}
});