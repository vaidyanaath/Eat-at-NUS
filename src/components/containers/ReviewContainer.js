import React from 'react';
import { TouchableOpacity, View, Image, StyleSheet } from 'react-native';

// import colors
import { colors } from '../../assets/colors';
import { SmallText } from '../texts/SmallText';

import { AirbnbRating } from 'react-native-ratings';

import { getTimeDifference } from '../../utils/getTimeDifference';

export const ReviewContainer = ({ author, review, rating, timeStamp }) => {
  return (
    <View style={cardStyles.container}>
      <View style={cardStyles.topSection}>
        <SmallText style={cardStyles.author}>{author}</SmallText>
        <AirbnbRating
          isDisabled={true}
          showRating={false}
          size={20}
          selectedColor={colors.primary}
          defaultRating={rating}
        />
      </View>
      <View style={cardStyles.reviewSection}>
        <SmallText style={cardStyles.review}>{review}</SmallText>
      </View>
      <View style={cardStyles.bottomSection}>
        <SmallText style={cardStyles.timeStamp}>{getTimeDifference(timeStamp, Date.now())}</SmallText>
      </View>
    </View>
  );
};

const cardStyles = StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'center',
    width: '99%',
    minHeight: 125,
    paddingHorizontal: 25,
    paddingVertical: 15,
    marginVertical: 5,
    borderRadius: 20,
    backgroundColor: colors.white,
    shadowColor: '#000',
    shadowOffset: { width: 5, height: 5 },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,
    elevation: 3,
  },
  topSection: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 10,
    // backgroundColor: '#cefe',
  },
  author: {
    fontWeight: 'bold',
  },
  reviewSection: {
  },
  bottomSection: {
    marginVertical: 5,
    alignSelf: 'flex-end',
  },
  timeStamp: {
    color: colors.darkGray,
  }
});
