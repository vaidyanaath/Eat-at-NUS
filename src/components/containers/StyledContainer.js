import styled from 'styled-components/native';
import Constants from "expo-constants";
import { colors } from '../../assets/colors';

const StatusBarHeight = Constants.statusBarHeight;

export const StyledContainer = styled.View`
    flex: 1;
    flex-direction: column;
    width: 100%;
    height: 100%;
    padding: 15px;
    padding-top: ${ StatusBarHeight }px;
    background-color: ${ colors.bg };
`;
