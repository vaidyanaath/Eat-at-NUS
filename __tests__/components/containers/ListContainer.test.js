import React from 'react';
import renderer from 'react-test-renderer';

import { ListContainer } from '../../../src/components/containers/ListContainer';

describe('<ListContainer />', () => {
    
    it('Renders correctly', () => {
        const mockItem = { name: "", id: "", rating: "" };
        const tree = renderer.create(<ListContainer item = {mockItem} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});