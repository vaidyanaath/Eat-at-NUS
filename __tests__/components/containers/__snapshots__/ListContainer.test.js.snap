// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListContainer /> Renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "activeOpacity": 0.8,
      "backgroundColor": "#FFFFFF",
      "borderColor": "#F5F5F5",
      "borderRadius": 20,
      "borderWidth": 0,
      "elevation": 3,
      "flex": 1,
      "height": 100,
      "marginVertical": 5,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 0.22,
      "shadowRadius": 2.22,
      "width": "98%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomLeftRadius": 20,
          "borderTopLeftRadius": 20,
          "flex": 1,
          "maxWidth": 140,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 20,
            "flex": 1,
            "resizeMode": "cover",
            "width": 125,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderBottomRightRadius": 20,
          "borderTopRightRadius": 20,
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2C365A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 23,
            },
            Object {
              "fontSize": 20,
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2C365A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 23,
            },
            Object {
              "flex": 2,
              "fontSize": 15,
            },
          ]
        }
      >
        
         Km away
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#FFB81C",
            "borderRadius": 3,
            "paddingHorizontal": 4,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 15,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
